src/components/JTextArea/JTextArea.js

```diff
@@ -28,37 +44,91 @@ function heightCalc({ currentTarget: target }: SyntheticEvent<HTMLTextAreaElemen
   }
 }
 
-export class JTextArea extends PureComponent<Props, void> {
+export class JTextArea extends PureComponent<Props, State> {
   static defaultProps = {
-    onBlur: undefined,
-    onFocus: undefined,
-    theme: 'blue',
+    onChange: noop,
+    onBlur: noop,
+    onFocus: noop,
+    theme: 'white',
     className: undefined,
+    label: undefined,
+    id: undefined,
+    value: undefined,
+    rows: 1,
+    error: false,
+    disabled: false,
+  }
+
+  constructor(props: Props) {
+    super(props)
+
+    this.state = {
+      focused: String(this.props.value).length > 0,
+    }
   }
 
-  handleChange = (fieldEvent: SyntheticEvent<HTMLTextAreaElement>): void => {
-    heightCalc(fieldEvent)
-    this.props.onChange(fieldEvent)
+  handleChange = (event: SyntheticEvent<HTMLTextAreaElement>): void => {
+    heightCalc(event)
+    this.props.onChange(event)
+  }
+
+  handleBlur = (event: SyntheticEvent<HTMLTextAreaElement>): void => {
+    if (String(this.props.value).length <= 0) {
+      this.setState({ focused: false })
+    }
+
+    this.props.onBlur(event)
+  }
+
+  handleFocus = (event: SyntheticEvent<HTMLTextAreaElement>): void => {
+    this.setState({ focused: true })
+    this.props.onFocus(event)
   }
 
   render() {
     const omitedProps = omit(this.props, [
-      'children',
       'onChange',
+      'children',
       'theme',
+      'label',
+      'id',
       'className',
+      'value',
+      'rows',
+      'placeholder',
+      'error',
     ])
 
+    const elementID: string = kebabCase(this.props.label) || undefined
+
     return (
-      <textarea
-        {...omitedProps}
+      <div
+        id={this.props.id}
         className={classNames(
           jTextAreaStyle.core,
           jTextAreaStyle[this.props.theme],
+          this.props.error && jTextAreaStyle.error,
+          this.state.focused && jTextAreaStyle.focused,
+          this.props.disabled && jTextAreaStyle.disabled,
           this.props.className,
         )}
-        onChange={this.handleChange}
-      />
+      >
+        {elementID && (
+          <label className={classNames(jTextAreaStyle.label)} htmlFor={elementID}>
+            {this.props.label}
+          </label>)}
+        <textarea
+          {...omitedProps}
+          onChange={this.handleChange}
+          onBlur={this.handleBlur}
+          onFocus={this.handleFocus}
+          id={elementID}
+          className={classNames(jTextAreaStyle.input)}
+          value={this.props.value}
+          rows={this.props.rows}
```

Comment:
seems value & rows could be passed with `omitedProps`

Reply:
It make sense, thank you